hooks/pre-commit

```diff
@@ -53,6 +53,22 @@ else
 fi
 echo "${reset}"
 
+allfiles=($(git diff --cached --name-only --diff-filter ACM | grep -v -e "third_party" -e "Godeps"))
```

Comment:
I see this pattern a lot. Would it make sense to get the list from git once and then apply subfilters? Probably not this PR.